src/main/java/bio/terra/common/flagsmith/FlagsmithService.java

```diff
@@ -57,10 +50,53 @@ public Optional<Boolean> isFeatureEnabled(String feature) {
       LOGGER.warn("Feature {} not found in {}", feature, flagsmithProperties.getApiUrl(), e);
     } catch (Exception e) {
       LOGGER.warn("Failed to fetch feature {}", feature, e);
+      throw e;
     }
     return Optional.empty();
   }
 
+  /**
+   * Get feature's value formatted as JSON.
+   *
+   * <p>If Flagsmith is unavailable, feature does not exist or the feature value does not exist,
+   * return {@code Optional.empty()}.
+   */
+  public <T> Optional<T> getFeatureValueJson(String feature, Class<T> clazz) throws Exception {
```

Comment:
nit: should this just be named getFeatureValue? since the type is controlled by clazz